share/smack/prelude.py

```diff
@@ -0,0 +1,290 @@
+import itertools
+
+class Attribute:
+  def __init__(self, attribute, attribute_arg = None):
+    self.attribute = attribute
+    self.attribute_arg = attribute_arg
+
+  def toStr(self):
+    attribute_str = '{:' + self.attribute
+    if self.attribute_arg is not None:
+      attribute_str += ' \"' + self.attribute_arg + '\"'
+    attribute_str += '}'
+    return attribute_str
+
+class Argument:
+  def __init__(self, name, type):
+    self.name = name
+    self.type = type
+
+  def toStr(self):
+    return self.name + ': ' + self.type
+
+class Function:
+  def __init__(self, name, attributes = [], arguments = [], return_type = None, body = None):
+    self.name = name
+    self.attributes = attributes
+    self.arguments = arguments
+    self.return_type = return_type
+    self.body = body
+
+  def toStr(self):
+    function_str = 'function'
+    for attribute in self.attributes:
+      function_str += ' ' + attribute.toStr()
+    function_str += ' ' + self.name + '('
+    for i, argument in enumerate(self.arguments):
+      function_str += argument.toStr()
+      if i < len(self.arguments) - 1:
+        function_str += ', '
+    function_str += ')'
+    if self.return_type is not None:
+      function_str += ' returns (' + self.return_type + ')'
+    if self.body is not None:
+      function_str += ' {'+ self.body + '}'
+    else:
+      function_str += ';'
+    function_str += '\n'
+    return function_str
+
+
+def int(bit_width):
+  return 'i' + str(bit_width)
+
+def bv(bit_width):
+  return 'bv' + str(bit_width)
+
+def uninterpreted_unary_op(bit_width, base_name):
+  return Function(
+    name = base_name + '.' + int(bit_width),
+    arguments = [Argument(name = 'i', type = int(bit_width))],
+    return_type = int(bit_width),
+  )
+
+def uninterpreted_binary_op(bit_width, base_name):
+  return Function(
+    name = base_name + '.' + int(bit_width),
+    arguments = [Argument(name = 'i1', type = int(bit_width)), Argument(name = 'i2', type = int(bit_width))],
+    return_type = int(bit_width)
+  )
+
+def inline_binary_op(bit_width, base_name, body):
+  return Function(
+    name = base_name + '.' + int(bit_width),
+    attributes = [Attribute(attribute = 'inline')],
```

Comment:
No need to use keyword arguments with a single argument which is obviously the attribute name. Just write `Attribute('inline')`.